samples/custom_dispatch/cpu/mlp_plugin/CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+if(NOT IREE_TARGET_BACKEND_LLVM_CPU OR
+   NOT IREE_HAL_DRIVER_LOCAL_SYNC OR
+   NOT IREE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF)
+  return()
+endif()
+
+# system-library plugin mechanism using the system dynamic library loader.
+if(IREE_HAL_EXECUTABLE_PLUGIN_SYSTEM_LIBRARY)
+
+  
+add_library(iree_samples_custom_dispatch_cpu_mlp_plugin SHARED
+  mlp_plugin.c
+)
+target_include_directories(iree_samples_custom_dispatch_cpu_mlp_plugin
+  PRIVATE
+    ${IREE_SOURCE_DIR}/runtime/src/
+)
+
+# NOTE: this is only required because we want this sample to run on all
+# platforms without needing to change the library name (libfoo.so/foo.dll).
+set_target_properties(iree_samples_custom_dispatch_cpu_mlp_plugin
+  PROPERTIES
+    WINDOWS_EXPORT_ALL_SYMBOLS ON
+    PREFIX ""
+    OUTPUT_NAME "mlp_plugin"
+)
+
+add_dependencies(iree-sample-deps
+  iree_samples_custom_dispatch_cpu_mlp_plugin
+  iree_samples_custom_dispatch_cpu_system_plugin)
+
+
+iree_lit_test_suite(
+  NAME
+    mlp_example
+  SRCS
+    "mlp.mlir"
```

Comment:
This test is failing on macos

Early discussion here: https://discord.com/channels/689900678990135345/1062405112292712499/1206999351038050406

Sample logs: https://github.com/openxla/iree/actions/runs/7882950336/job/21509014526#step:5:2662
```
1319/1334 Test #1801: iree/samples/custom_dispatch/cpu/mlp_plugin/mlp.mlir.test ..................................................***Failed    0.79 sec
-- Testing: 1 tests, 1 workers --
FAIL: IREE :: custom_dispatch/cpu/mlp_plugin/mlp.mlir (1 of 1)
******************** TEST 'IREE :: custom_dispatch/cpu/mlp_plugin/mlp.mlir' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 1: iree-compile --iree-preprocessing-transform-spec-filename=/work/samples/custom_dispatch/cpu/mlp_plugin/mlp_spec.mlir  /work/samples/custom_dispatch/cpu/mlp_plugin/mlp.mlir |  iree-run-module --device=local-sync      --executable_plugin=$IREE_BINARY_DIR/samples/custom_dispatch/cpu/mlp_plugin/mlp_plugin$IREE_DYLIB_EXT      --module=-      --function=mlp_invocation      --input="2x2xf32=[[2.0, 2.0], [-2.0, -2.0]]"      --input="2x2xf32=[[3.0 -3.0], [3.0, -3.0]]"
+ iree-compile --iree-preprocessing-transform-spec-filename=/work/samples/custom_dispatch/cpu/mlp_plugin/mlp_spec.mlir /work/samples/custom_dispatch/cpu/mlp_plugin/mlp.mlir
+ iree-run-module --device=local-sync --executable_plugin=/work/build-arm64/samples/custom_dispatch/cpu/mlp_plugin/mlp_plugin.so --module=- --function=mlp_invocation '--input=2x2xf32=[[2.0, 2.0], [-2.0, -2.0]]' '--input=2x2xf32=[[3.0 -3.0], [3.0, -3.0]]'
<vm>:0: UNAVAILABLE; none of the executable binaries in the module are supported by the runtime; 
[ 0] bytecode example.__init:260 /work/samples/custom_dispatch/cpu/mlp_plugin/mlp.mlir:28:1; creating VM context; creating run context
```

Reply:
By definition, this test is only coded for x86 (literally embeds an x86 binary in it). So the error is the correct outcome on ARM. I can never remember with all of these lit-ish variants how we mark for only certain archs...

Reply:
Probably need to add a `requires-x86_64-target` label and include that at on x86 I guess. Or to unbreak HEAD, add a cmake `if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")`

Reply:
Oh sorry! I'll try to find the guard for cmake flag here.